src/vsr.zig

```diff
@@ -1336,19 +1339,18 @@ const ViewChangeHeadersArray = struct {
                 // Backup: these headers are stored in the superblock's vsr_headers.
                 switch (chain) {
                     .chain_sequence => {},
-                    // Gaps are due to either:
-                    // - entries before checkpoint, which are not repaired, or
+                    // Gaps are due to:
                     // - backup missed prepares and has not repaired headers. (Immediately after
                     //   receiving a start_view this is not a concern, but the view_durable_update()
                     //   may be delayed if another is in progress).
                     .chain_view, .chain_gap => {
-                        assert(op <= options.op_checkpoint or !current.log_view_primary);
+                        assert(!current.log_view_primary);
                         break;
                     },
                     // Breaks are due to:
-                    // - entries before checkpoint, which are not repaired
+                    // - entries before checkpoint, which are not repaired by backups
                     .chain_break => {
-                        assert(op <= options.op_checkpoint);
+                        assert(!current.log_view_primary and op <= options.op_checkpoint);
                         break;
```

Comment:
I have a follow-up PR for this one (wip) that will remove this whole function — `op_repair_min()` doesn't actually make view-change header generation any easier, but it did show me a simpler implementation.